### PR TITLE
Joker limit

### PR DESCRIPTION
**Pull request by MarieDugoua**

### CODE REVIEW

File: tests/game.test.ts [hunk 1]
```diff
@@ -34,16 +34,29 @@ describe('The test environment', () => {
...
+
+        game.addPlayer('Pet')
+        game.addPlayer('Ed')
+
+        game.roll(3);
+        game.useJoker();
+
+        expect(consoleSpy.content).not.to.include("has used their Joker");
+    });
 });
```
Comment (Saiiga) [inline tests/game.test.ts:298]: Add a test when joker is used

File: src/playerPool.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+
 export class PlayerPool {
 
+    private console: IConsole;
+
     private _players: Array<string> = [];
     private _places: Array<number> = [];
     private _purses: Array<number> = [];
     private _inPenaltyBox: Array<boolean> = [];
+    private _usedJoker: Array<boolean> = [];
```
Comment (Saiiga) [inline src/playerPool.ts:10]: Change the name to make it understable that was a list

File: tests/game.test.ts [hunk 3]
```diff
@@ -46,6 +46,21 @@ describe('The test environment', () => {
...
+
+    it("should not use joker if player has no joker", () => {
+        const consoleSpy = new ConsoleSpy();
+        const game = new Game(consoleSpy);
+
+        game.addPlayer('Pet')
+        game.addPlayer('Ed')
+
+        game.roll(3);
+        game.useJoker();
```
Comment (Saiiga) [inline tests/game.test.ts]: can't work

Comment (Saiiga) [reply tests/game.test.ts]: verify : 
number of coin
if joker was used

File: src/playerPool.ts [hunk 4]
```diff
@@ -95,7 +103,25 @@ export class PlayerPool {
...
+            this._usedJoker[this.currentPlayer] = true;
+            this.console.log(this.players[this.currentPlayer] + " has used their Joker.");
+        }
+    }
+
+    public hasCurrentPlayerUsedJoker(): boolean {
+        return this._usedJoker[this.currentPlayer];
+    }
+
+    public currentPlayerUseJoker(): void {
```
Comment (Saiiga) [inline src/playerPool.ts:134]: why ? not uselesss ?

File: src/playerPool.ts [hunk 5]
```diff
@@ -1,13 +1,19 @@
+import {IConsole} from "./Utils/IConsole";
+
 export class PlayerPool {
 
+    private console: IConsole;
+
     private _players: Array<string> = [];
```
Comment (Saiiga) [inline src/playerPool.ts]: rename to "playersName"

File: src/playerPool.ts [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
     private _places: Array<number> = [];
     private _purses: Array<number> = [];
     private _inPenaltyBox: Array<boolean> = [];
+    private _usedJoker: Array<boolean> = [];
     private _currentPlayer: number = 0;
```
Comment (Saiiga) [inline src/playerPool.ts:11]: rename to currentPlayerIndex

**Comment by Lotfihihi** [comment]
manque beaucoup de choses du master actuel

File: src/game-runner.ts [hunk 7]
```diff
@@ -11,13 +11,14 @@ export class GameRunner {
         do {
 
             game.roll(Math.floor(Math.random() * 6) + 1);
-        
+            game.useJoker();
```
Comment (Saiiga) [inline src/game-runner.ts]: Remove the hardcoded call

File: src/game.ts [hunk 8]
```diff
@@ -5,7 +5,8 @@ import {TooManyPlayerError} from "./errors/TooManyPlayerError";
 import {PlayerPool} from "./playerPool";
 
 export class Game {
-    private currentPlayer: number = 0;
+    private usedJoker: Array<boolean> = [];
```
Comment (Saiiga) [inline src/game.ts]: Move this logic in PlayerPool

**Review by Saiiga — CHANGES_REQUESTED** [review]

**Review by Lotfihihi — APPROVED** [review]
OK

**Review by Saiiga — APPROVED** [review]

**Pull request merged**